### PR TITLE
[FEATURE] Improve attachments

### PR DESCRIPTION
**Pull request by schams-net**
This change extends the TYPO3 Forum extension by the following features.

* allow users to add **multiple** attachments by selecting more than one file in browser
* allow users to **delete** attachments (edit post → delete)

In addition, files involved in this change have been cleaned up to follow PSR-2 coding guidelines and some missing comments/annotations have been added.

### CODE REVIEW

File: Resources/Private/Partials/Bootstrap/Post/Form.html [hunk 1]
```diff
@@ -1,5 +1,16 @@
 {namespace mmf=Mittwald\Typo3Forum\ViewHelpers}
 {namespace b=Mittwald\Typo3Forum\ViewHelpers\Bootstrap}
+<script type="text/javascript">
+	$(document).ready(function() {
+	function validate() {
+		var file_size = $('#att')[0].files[0].size / 1000;
+		if (file_size > 2000) {
+			alert("Uploaded file(s) size should be less than 2MB");
```
Comment (DavidBruchmann) [inline Resources/Private/Partials/Bootstrap/Post/Form.html]: Why to MB? That's a limit that should be looked up in php.ini and could be 200MB too.

File: Resources/Private/Partials/Bootstrap/Post/Form.html [hunk 2]
```diff
@@ -29,17 +40,36 @@
...
 						<mmf:format.fileSize>{attachment.filesize}</mmf:format.fileSize>
 					</div>
 				</f:for>
 			</f:if>
-			<f:form.upload name="attachments[0]"/>
 		</b:form.row>
+
+		<b:form.row>
+			<f:form.upload name="attachments[0]" multiple="1" id="att" />
+			<a href="javascript:void(0);" onclick="clearUpload();" style="display:block;float:left;font-size: 24px;"><i class="fa fa-times" aria-hidden="true" style="fontsize:18px;color:#000;"></i></a>
```
Comment (DavidBruchmann) [inline Resources/Private/Partials/Bootstrap/Post/Form.html]: Dependency to font-awesome? That's in BE where it's available anyway, right?

File: Resources/Private/Templates/Bootstrap/Topic/New.html [hunk 3]
```diff
@@ -5,7 +5,18 @@
 
 
 <f:section name="main">
+<script type="text/javascript">
+	$(document).ready(function() {
 
+	function validate() {
+		var file_size = $('#att')[0].files[0].size / 1000;
+		if (file_size > 2000) {
+			alert("Uploaded file(s) size should be less than 2MB");
```
Comment (DavidBruchmann) [inline Resources/Private/Templates/Bootstrap/Topic/New.html]: see above, take limit from php.ini

**Comment by DavidBruchmann** [comment]
Should `[TASK] Allow deleting attachments` depend on rights perhaps, that might be adjustable in BE?

**Comment by DavidBruchmann** [comment]
Has "Add missing comments/annotations" any impact?

File: Classes/Service/AttachmentService.php [hunk 4]
```diff
@@ -25,30 +30,37 @@ public function initAttachments(array $attachments)
...
-            $attachmentObj->setMimeType($mime_type);
-
-            //Create dir if not exists
-            $tca = $attachmentObj->getTCAConfig();
-            $path = $tca['columns']['real_filename']['config']['uploadfolder'];
-            if (!file_exists($path)) {
-                mkdir($path, 0777, true);
-            }
+        foreach ($attachments as $attachment) {
+            foreach ($attachment as $attachmentID => $eachAttachment) {
```
Comment (DavidBruchmann) [inline Classes/Service/AttachmentService.php]: looks strange this double-loop, is it required?

File: Classes/Controller/PostController.php [hunk 5]
```diff
@@ -241,6 +243,7 @@ public function newAction(Topic $topic, Post $post = null, Post $quote = null)
         $this->view->assignMultiple([
             'topic' => $topic,
             'post' => $post,
+            'posts' => $posts,
```
Comment (DavidBruchmann) [inline Classes/Controller/PostController.php]: What's the purpose about `$posts`. IMO `$post` is transferred for being able to quote. Looks like a feature that is not covered by my horizon.

File: Resources/Private/Partials/Bootstrap/Post/Form.html (hunk 1 above)
Comment (DavidBruchmann) [reply Resources/Private/Partials/Bootstrap/Post/Form.html]: Limit in the extension (or of TYPO3) is thinkable and useful additionally I think, if it's not too much work.

**Comment by DavidBruchmann** [comment]
Can https://github.com/DavidBruchmann/typo3-forum-documentation be populated too about the new feature?

File: Resources/Private/Partials/Bootstrap/Post/Form.html (hunk 1 above)
Comment (DavidBruchmann) [reply Resources/Private/Partials/Bootstrap/Post/Form.html]: Also see #44

File: Resources/Private/Templates/Bootstrap/Topic/New.html (hunk 3 above)
Comment (schams-net) [reply Resources/Private/Templates/Bootstrap/Topic/New.html]: Solved in next pull request.

File: Classes/Controller/PostController.php [hunk 6]
```diff
@@ -282,10 +282,33 @@ public function editAction(Post $post)
      */
     public function updateAction(Post $post, array $attachments = [])
     {
+        // @TODO use proper Extbase code to retrieve POST data
+        $selectedFiles = $_POST['tx_typo3forum_pi1']['deleteAttachment'];
```
[the remark this answers is not in the document]
Comment (schams-net) [reply Classes/Controller/PostController.php]: Solved in next pull request.

File: Classes/Service/AttachmentService.php (hunk 4 above)
Comment (schams-net) [reply Classes/Service/AttachmentService.php]: under investigation.

File: Classes/Controller/PostController.php (hunk 5 above)
Comment (schams-net) [reply Classes/Controller/PostController.php]: under investigation.

**Comment by DavidBruchmann** [comment]
Thanks a lot for your work!

File: Classes/Controller/PostController.php (hunk 5 above)
Comment (DavidBruchmann) [reply Classes/Controller/PostController.php]: Even merged already, I still like to get an impression of the purpose here ...

**Pull request merged**